odl/discr/partition.py

```diff
@@ -766,6 +767,132 @@ def uniform_partition_fromgrid(grid, begin=None, end=None):
     return RectPartition(IntervalProd(begin_vec, end_vec), grid)
 
 
+def uniform_partition_options(begin=None, end=None, num_nodes=None,
+                              cell_sides=None):
+    """Return a partition with equally sized cells.
+
+    This function offers more flexibility over `uniform_discr` in that
+    any combination of 3 or 4 out of 4 possible init parameters can be
+    given, and the (possibly) missing one is calculated.
+
+    Parameters
+    ----------
+    begin, end : float or array-like, optional
+        Vectors defining the begin end end points of an `IntervalProd`
+        (a rectangular box). For one-dimensional partitions, single
+        floats can be provided. ``None`` entries mean "compute the value".
+    num_nodes : int or sequence of int, optional
+        Number of nodes per axis. For 1d intervals, a single integer
+        can be specified. ``None`` entries mean "compute the value".
+    cell_sides : float or array-like, optional
+        Side length of the partition cells per axis. For 1d intervals,
+        a single integer can be specified. ``None`` entries mean
+        "compute the value".
+
```

Comment:
Addition: With `nodes_on_bdry` added, the argument for keeping this function separate from `uniform_partition` us much weaker. I'll reconsider.